californium-core/src/test/java/org/eclipse/californium/core/test/lockstep/LockstepEndpoint.java

```diff
@@ -212,9 +259,61 @@ public void setDestination(InetSocketAddress destination) {
 		this.destination = destination;
 	}
 
+	/**
+	 * Receive next message.
+	 * 
+	 * Apply smart deduplication based on {@link #lastIncomingMessage} and the
+	 * MID, if the repeated MID is not expected. If no next message arrives,
+	 * reports an assert.
+	 * 
+	 * <code>
+	 *    ... expectRequest.storeMID("A").type(CON) ...
+	 * 
+	 *        // wait until retransmission
+	 * 
+	 *        // OK, expecting the MID suppresses deduplication
+	 *    ... expectRequest.sameMID("A").type(CON) ... 
+	 *        // will fail, not expecting the MID, 
+	 *        // the retransmission would be dropped by deduplication
+	 *    ... expectRequest.type(CON)... 
+	 * 
+	 * </code>
+	 * 
+	 * MID expectations are based on {@link MessageExpectation#mid(int)},
+	 * {@link MessageExpectation#sameMID(String)} or
+	 * {@link MessageExpectation#sameBoth(String)}.
+	 * 
+	 * @param midExpectation MID expectation
+	 * @return next received message
+	 * @throws InterruptedException if waiting for message is interrupted.
+	 */
+	public Message receiveNextMessage(MIDExpectation midExpectation) throws InterruptedException {
+		while (true) {
+			RawData raw = incoming.poll(2, TimeUnit.SECONDS); // or take()?
+			assertNotNull("Did not receive a message (but nothing)", raw);
```

Comment:
can we instead use something like `did not receive message within expected time frame (2 secs)`?